### DISCUSSION TITLE
How to create tabs for images

### DISCUSSION THREAD

**Discussion by Polo224** [community/community#131609] [New to GitHub]
Hi There, I am new to any of these coding and programming and I really have no idea how to do anything. I've been asked to add multiple images on a page where user can tab through according to what they want to see. Below you can find a sample of an existing page and below the one I created that does not work. I basically copy pasted the whole thing on a different dm document and changed just what I thought I needed to change.

Original:
Tabs groupId="how-to-add-a-conquest-account"
TabItem value="s1" label="Desktop Home Page" default
img src="/img/Home_Page.png" /
/TabItem
TabItem value="s2" label="Mobile Device Home Page"
img src="/img/Conquest_Mobile_homepage.jpg" /
/TabItem
/Tabs

New:
Tabs groupId="how-to-add-a-conquest-account"
TabItem value="s1" label="Desktop" default
img src="/img/Desktop_Tasks_Window.png" /
/TabItem
TabItem value="s2" label="Mobile Device"
img src="/img/Tasks_Menu.jpg" /
/TabItem
/Tabs

Can anyone please tell me if there is anything else I need to change or why this is not working on the new dm?

Thanks

**Comment by fruetalo182** [comment]
Try this:
```
"use client"

import { useState } from 'react'
import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs"
import Image from "next/image"

interface GalleryImage {
  id: string
  label: string
  src: string
}

const galleryImages: GalleryImage[] = [
  { id: "s1", label: "Desktop Home Page", src: "/img/Home_Page.png" },
  { id: "s2", label: "Mobile Device Home Page", src: "/img/Conquest_Mobile_homepage.jpg" },
]

export default function ImageGallery() {
  const [activeTab, setActiveTab] = useState(galleryImages[0].id)

  return (
    <Tabs value={activeTab} onValueChange={setActiveTab} className="w-full max-w-3xl mx-auto">
      <TabsList className="grid w-full grid-cols-2">
        {galleryImages.map((image) => (
          <TabsTrigger key={image.id} value={image.id}>
            {image.label}
          </TabsTrigger>
        ))}
      </TabsList>
      {galleryImages.map((image) => (
        <TabsContent key={image.id} value={image.id} className="mt-4">
          <div className="relative aspect-video w-full overflow-hidden rounded-lg">
            <Image
              src={image.src}
              alt={image.label}
              layout="fill"
              objectFit="cover"
            />
          </div>
        </TabsContent>
      ))}
    </Tabs>
  )
}
```
